pkg/setup/initialize.go

```diff
@@ -181,3 +197,21 @@ func GenerateK8sCerts(ctx context.Context, currentNamespaceClient kubernetes.Int
 
 	return nil
 }
+
+func updateSecret(ctx context.Context, currentNamespaceClient kubernetes.Interface, currentNamespace, vclusterName string, files map[string][]byte) error {
```

Comment:
This is a big race condition, multiple replicas can be at this function at the same time and will all one after another override this secret with new files